src/app/shared-services/location.service.ts

```diff
@@ -109,13 +109,15 @@ export class LocationService implements OnDestroy {
     this.locationUpdateSubscription = this.backgroundGeolocation
       .on(BackgroundGeolocationEvents.location)
       .subscribe(async ({ latitude, longitude, accuracy, speed, time }) => {
-        await this.db.upsertPoint(TRACKING_TRAJ_ID, {
+        await this.db.upsertPoint(Trajectory.trackingTrajectoryID, {
           latLng: [latitude, longitude],
           time: new Date(time),
           accuracy,
           speed,
         })
 
+        this.inf.generateUserInference()
```

Comment:
This is executed asynchronously, right? In that case, please add a comment noting that.

But are you sure we can run it asynchronously? [According to background location plugin docs](https://github.com/mauron85/cordova-plugin-background-geolocation#example), execution time is limited in this callback for iOS, and should be wrapped in a "transaction" like this:
```js
  BackgroundGeolocation.on('location', function(location) {
    // handle your locations here
    // to perform long running operation on iOS
    // you need to create background task
    BackgroundGeolocation.startTask(function(taskKey) {
      // execute long running task
      // eg. ajax post location
      // IMPORTANT: task has to be ended by endTask
      BackgroundGeolocation.endTask(taskKey);
    });
  });
```

Reply:
I'll need to take a closer look at this.

Potentially problematic: This probably triggers the usual background processing in iOS, which is pretty strictly controlled by the OS. In the use-case of the learning app, which may not to be actively used very much, the OS could rate our app as low-priority and thus give it very little to no background time for such tasks. The location updates on the other hand are triggered quite regularly, therefore the execution is currently done directly there. Will update on this asap.